lib/commands/content-size.js

```diff
@@ -0,0 +1,69 @@
+import _ from 'lodash';
+import {assertSimulator as _assertSimulator} from '../utils';
+import { errors } from 'appium/driver';
+
+const assertSimulator = _.partial(_assertSimulator, 'Content size ui command');
+
+const CONTENT_SIZE = [
+    'extra-small',
+    'small',
+    'medium',
+    'large',
+    'extra-large',
+    'extra-extra-large',
+    'extra-extra-extra-large',
+    'accessibility-medium',
+    'accessibility-large',
+    'accessibility-extra-large',
+    'accessibility-extra-extra-large',
+    'accessibility-extra-extra-extra-large',
+    'increment',
+    'decrement'
+];
+
+export default {
+  /**
+   * Sets content size for the given simulator.
+   *
+   * @since Xcode 15 (but lower xcode could have this command)
+   * @param {ContentSizeAction} size - The content size action to set. Acceptable value is
+   *                                   extra-small, small, medium, large, extra-large, extra-extra-large,
+   *                                   extra-extra-extra-large, accessibility-medium, accessibility-large,
+   *                                   accessibility-extra-large, accessibility-extra-extra-large,
+   *                                   accessibility-extra-extra-extra-large with Xcode 16.2.
+   * @throws {Error} if the current platform does not support content size appearance changes
+   * @this {XCUITestDriver}
+   */
+  async mobileSetContentSize(size) {
+    assertSimulator(this);
```

Comment:
maybe we could already return simulator instance from this typeguard to avoid the following typecast